quantum/process_keycode/process_leader.c

```diff
@@ -49,24 +49,24 @@ bool process_leader(uint16_t keycode, keyrecord_t *record) {
   // Leader key set-up
   if (record->event.pressed) {
     if (leading) {
-      if (timer_elapsed(leader_time) < LEADER_TIMEOUT) {
-#ifndef LEADER_KEY_STRICT_KEY_PROCESSING
+    //   if (timer_elapsed(leader_time) < LEADER_TIMEOUT) {
```

Comment:
This should not be removed..  It should be hidden behind a define, if anything. 

```suggestion
#ifndef LEADER_NO_TIMEOUT
      if (timer_elapsed(leader_time) < LEADER_TIMEOUT)
#endif
      {
```

Reply:
The thing is that line does not fulfill any purpose. Could you please explain to me what is this for? The timeout is already checked on `LEADER_DICTIONARY`

Reply:
But it does, though.  It prevents the leader key from resetting the leader sequence, if hit again during the sequence. 

I'm not sure that removing it is the best idea here.

Reply:
Hello @drashna, thanks for your answer. I still don't get how this check prevents the leader key from resetting the sequence. Would you mind explaining it to me a bit further? We can even find a more direct way of discussing it if you prefer.

Reply:
Basically, it only runs this code if the elapsed time is less than the leader timeout, eg if it's within the leader "time window".  

Hiding the check behind a define means that you can use either option (with a timeout or not).  
Removing the check means that you can ONLY use the infinite timeout.

Reply:
My point is that it is useless in both cases. But this asynchronous conversation is making it more difficult. Are you open to discuss this on discuss ? I think we have the same nick on that platform on the QMK channel.

Reply:
@drashna added